x-pack/elastic-agent/pkg/composable/providers/kubernetesleaderelection/config.go

```diff
@@ -0,0 +1,15 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+// TODO review the need for this
+// +build linux darwin windows
+
+package kubernetesleaderelection
+
+// Config for kubernetes_leaderelection provider
+type Config struct {
+	KubeConfig string `config:"kube_config"`
+	// Name of the leaderelection lease
+	LeaderLease string `config:"leader_lease"`
```

Comment:
So by default running Elastic Agent this is off. Is that the behavior we want? Being that will require a specific setting in the `providers` top-level key, and not something that could then be used by Fleet at the moment.

I think that we should have this enabled by default, by having it have a default value.

Reply:
Agree, I will add it.